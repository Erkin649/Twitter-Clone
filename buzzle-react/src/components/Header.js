import React from 'react';
import { Link } from 'react-router-dom';
import UserToolbar from '../containers/UserToolbar.js';
import SearchBar from '../containers/SearchBar.js';


class Header extends React.Component{
    render(){
        const showUserToolbar = this.props.showUserToolbar;

        return(
            <header>
                <nav className="navbar navbar-dark bg-primary">
                    <Link className="navbar-brand" to="/">
                        <h1>Buzzle</h1>
                    </Link>
                    <SearchBar />
                    {showUserToolbar && <UserToolbar />}
                </nav>
            </header>
        )
    }
}
export default Header;

// const Header = ({ showUserToolbar }) =>
//     <header>
//         <nav className="navbar navbar-dark bg-primary">
//             <Link className="navbar-brand" to="/">
//                 <h1>Buzzle</h1>
//             </Link>
//             <form className="form-inline">
//                 <input className="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search" />
//                 <button className="btn btn-outline-light my-2 my-sm-0" type="submit">Search</button>
//             </form>
//             {showUserToolbar && <UserToolbar />}
//         </nav>
//     </header>

