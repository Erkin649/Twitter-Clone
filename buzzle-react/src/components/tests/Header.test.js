import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Header from '../Header.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router><Header /></Router>, div);
});

it('renders correctly and equal to snapshot', () => {
    const tree = renderer.create(<Router><Header /></Router>)
                         .toJSON();
  
    expect(tree).toMatchSnapshot();
});