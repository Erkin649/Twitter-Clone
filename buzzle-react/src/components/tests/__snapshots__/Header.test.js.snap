// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and equal to snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-dark bg-primary"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <h1>
        Buzzle
      </h1>
    </a>
  </nav>
</header>
`;
